extensions/jms/src/main/java/org/mule/extensions/jms/api/connection/JmsTransactionalConnection.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extensions.jms.api.connection;
+
+import static org.mule.extensions.jms.TransactionStatus.NONE;
+import static org.mule.extensions.jms.TransactionStatus.STARTED;
+import static org.slf4j.LoggerFactory.getLogger;
+import org.mule.extensions.jms.JmsExtension;
+import org.mule.extensions.jms.JmsSessionManager;
+import org.mule.extensions.jms.internal.support.JmsSupport;
+import org.mule.runtime.extension.api.connectivity.TransactionalConnection;
+
+import org.slf4j.Logger;
+
+import java.util.Optional;
+
+import javax.jms.Connection;
+import javax.jms.JMSException;
+import javax.jms.Session;
+
+/**
+ * Implementation of the {@link JmsConnection} which implements {@link TransactionalConnection} for Transaction Support
+ * in the {@link JmsExtension}
+ *
+ * @since 4.0
+ */
+public final class JmsTransactionalConnection extends JmsConnection implements TransactionalConnection {
```

Comment:
Why can't the JmsConnection implement the TransactionalConnection instead of this hierarchy? 
We are always using the JmsTransactionalConnection, that's what the provider creates and what every processor uses.